pulse_otel/util.py

```diff
@@ -135,3 +164,21 @@ def _is_endpoint_reachable(endpoint_url: str, timeout: int = 3) -> bool:
     except Exception as e: # Catch any other unexpected errors during the check
         print(f"Warning: An unexpected error occurred while checking OTel endpoint reachability for {endpoint_url}: {e}")
         return False
+
+def add_session_id_to_span_attributes(**kwargs):
+	"""
+	Extracts the session ID from the `baggage` header in the provided kwargs and sets it as an association property for tracing.
+	Args:
+		kwargs (dict): A dictionary that may contain a 'headers' key with HTTP headers.
+	Returns:
+		None
+	"""
+	session_id = extract_session_id(**kwargs) or extract_session_id_from_body(**kwargs)
+
+	if not session_id:
+		session_id = str(random.randint(10**15, 10**16 - 1))
```

Comment:
is there a reason we should be generating a sessionID here? if not for predictability for now IMO the SDK should only be responsible for extracting and threading the session ID